service/view_building_coordinator.cc

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (C) 2025-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+ */
+
+#include <exception>
+#include <iterator>
+#include <ranges>
+#include <seastar/core/coroutine.hh>
+
+#include "cql3/query_processor.hh"
+#include "db/schema_tables.hh"
+#include "db/system_keyspace.hh"
+#include "mutation/canonical_mutation.hh"
+#include "schema/schema_fwd.hh"
+#include "seastar/core/loop.hh"
+#include "seastar/coroutine/maybe_yield.hh"
+#include "service/raft/group0_state_machine.hh"
+#include "service/raft/raft_group0.hh"
+#include "service/raft/raft_group0_client.hh"
+#include "utils/assert.hh"
+#include "utils/log.hh"
+#include "seastar/core/abort_source.hh"
+#include "seastar/core/with_scheduling_group.hh"
+#include "service/migration_manager.hh"
+#include "replica/database.hh"
+#include "view_info.hh"
+
+#include "service/view_building_coordinator.hh"
+
+static logging::logger vbc_logger("view_building_coordinator");
+
+namespace service {
+
+namespace vbc {
+
+view_building_coordinator::view_building_coordinator(abort_source& as, replica::database& db, raft_group0& group0, db::system_keyspace& sys_ks, const topology_state_machine& topo_sm) 
+    : _db(db)
+    , _group0(group0)
+    , _sys_ks(sys_ks)
+    , _topo_sm(topo_sm)
+    , _as(as) 
+{}
+
+future<group0_guard> view_building_coordinator::start_operation() {
+    auto guard = co_await _group0.client().start_operation(_as);
+    co_return std::move(guard);
+}
+
+future<> view_building_coordinator::await_event() {
+    _as.check();
+    co_await _cond.when();
+    vbc_logger.debug("event awaited");
+}
+
+future<view_building_coordinator::vbc_state> view_building_coordinator::load_coordinator_state() {
+    auto tasks = co_await _sys_ks.get_view_building_coordinator_tasks();
+    co_return vbc_state {
+        .tasks = std::move(tasks)
+    };
+}
+
+table_id view_building_coordinator::get_base_id(const view_name& view_name) {
+    return _db.find_schema(view_name.first, view_name.second)->view_info()->base_id();
+}
+
+future<> view_building_coordinator::run() {
+    auto abort = _as.subscribe([this] noexcept {
+        _cond.broadcast();
+    });
+
+    while (!_as.abort_requested()) {
+        vbc_logger.debug("coordinator loop iteration");
+        try {
+            auto state_opt = co_await update_coordinator_state();
+            if (!state_opt) {
+                // If state_opt is nullopt, it means there was work to do and the state has changed.
+                continue;
+            }
+            // TODO: Do actual work, send RPCs to build a particular view's range
+            co_await await_event();
+        } catch (...) {
+            
```

Comment:
?

Reply:
Error handling is added in a later commit, added TODO comment